tensorflow/lite/micro/kernels/softmax.cc

```diff
@@ -72,12 +99,46 @@ TfLiteStatus CalculateSoftmaxParams(TfLiteContext* context,
 
 }  // namespace
 
+void* SoftmaxInit(TfLiteContext* context, const char* buffer, size_t length) {
+  void* raw = nullptr;
+  context->AllocatePersistentBuffer(context, sizeof(SoftmaxOpData), &raw);
```

Comment:
Could you break this allocation up so that exp_lut and one_over_one_plus_x_lut are allocated separately, and only if the type is int16? Otherwise the tables will take up 2KB of RAM even for types that aren't using them. Something like

```
struct SoftmaxOpData {
...
int16_t exp_lut*;
int16_t one_over_one_plus_x_lut*;
};
...
  void* raw_op_data = nullptr;
  context->AllocatePersistentBuffer(context, sizeof(SoftmaxOpData), &raw_op_data);
  SoftmaxOpData* op_data = reinterpret_cast<SoftmaxOpData*>(raw_op_data);
  if (input->type == kTfLiteInt16) {
    void* raw_exp_lut;
    context->AllocatePersistentBuffer(context, sizeof(int16_t) * size_of_lut, &raw_exp_lut);
    op_data->exp_lut = reinterpret_cast<int16_t*>(raw_exp_lut);
   ...
```

Reply:
Done. That's a good point!